Storage/Passwords.swift

```diff
@@ -70,9 +65,7 @@ public class Password : Equatable {
     }
 
     public class func fromScript(url: NSURL, script: [String: String]) -> Password {
-        let site = Site(url: getPasswordOrigin(url.absoluteString!)!, title: "")
-
-        let pswd = Password(site: site, username: script["username"]!, password: script["password"]!)
+        let pswd = Password(hostname: getPasswordOrigin(url.absoluteString!)!, username: script["username"]!, password: script["password"]!)
```

Comment:
Too many exclamation marks. I think specially for things that come from JS we should do strict validation. Right now if usernam or password is missing or nil, the app hard crashes.

For Reading View I use wrap the data in a struct with a failable initializer. More code but it is impossible to get bad data. https://github.com/mozilla/firefox-ios/blob/master/Client/Frontend/Reader/ReaderMode.swift#L105